app/models/threads.js

```diff
@@ -308,12 +308,38 @@
             if (updates.distribution && 'expression' in updates.distribution &&
                 updates.distribution.expression != this.get('distribution')) {
                 const dist = updates.distribution.expression;
+                const oldDist = this.get('distribution');
                 const normalized = await F.atlas.resolveTagsFromCache(dist);
+                const oldNormalized = await F.atlas.resolveTagsFromCache(oldDist); 
                 if (normalized.universal !== updates.distribution.expression) {
                     F.util.reportError('Non-universal expression sent by peer',
                                        {distribution: dist});
                 }
-                this.addNotice("Distribution Changed", normalized.pretty);
+                let added;
+                let diffIncludedTags;
+                let diffExcludedTags;
+                var newIncludedTags = new F.util.ESet(normalized.includedTagids);                
+                var oldIncludedTags = new F.util.ESet(oldNormalized.includedTagids);
+                var newExcludedTags = new F.util.ESet(normalized.excludedTagids);                
+                var oldExcludedTags = new F.util.ESet(oldNormalized.excludedTagids);
+                if (newIncludedTags.size > oldIncludedTags.size || newExcludedTags.size > oldExcludedTags.size) {
```

Comment:
Sorry, you are still going to miss important cases because a set may not change size while still gaining and losing members. A cleaner approach might be to not screen by set sizes and simply evaluate the set math to get your overall added and removed tags:
* added = (oldExcluded - newExcluded) + (newIncluded - oldIncluded)
* removed = (newExcluded - oldExcluded) + (oldIncluded - newIncluded)